src/overlays/actors/ovl_En_Stream/z_en_stream.c

```diff
@@ -29,19 +31,117 @@ const ActorInit En_Stream_InitVars = {
     (ActorFunc)EnStream_Update,
     (ActorFunc)EnStream_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Stream/func_80B0B7A0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Stream/EnStream_Init.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 20, ICHAIN_STOP),
+};
+
+void EnStream_SetupAction(EnStream* this, EnStreamActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnStream_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnStream* this = THIS;
+
+    this->unk_150 = thisx->params & 0xFF;
+    Actor_ProcessInitChain(thisx, sInitChain);
+    if ((this->unk_150 != 0) && (this->unk_150 == 1)) {
+        thisx->scale.y = 0.01f;
+    }
+    EnStream_SetupAction(this, &EnStream_WaitForPlayer);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Stream/EnStream_Destroy.s")
+void EnStream_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
+#ifdef NONMATCHING
+// regalloc differences, checks if the player is range of the vortex
+s32 func_80B0B81C(PosRot* vortexPosRot, PosRot* playerPosRot, Vec3f* posRotDifference, f32 vortexYScale) {
+    s32 ret = 0;
+    f32 smallConstant = 28;
+    f32 lowerBounds = 0 * vortexYScale * 50;
+    f32 upperBounds = 160 * vortexYScale * 50;
+    f32 xzDist;
+
+    posRotDifference->x = playerPosRot->pos.x - vortexPosRot->pos.x;
+    posRotDifference->y = playerPosRot->pos.y - vortexPosRot->pos.y;
+    posRotDifference->z = playerPosRot->pos.z - vortexPosRot->pos.z;
+    xzDist = sqrtf(SQ(posRotDifference->x) + SQ(posRotDifference->z));
+    if (lowerBounds <= posRotDifference->y && posRotDifference->y <= upperBounds) {
+        posRotDifference->y = posRotDifference->y - lowerBounds;
+        if (xzDist <= (((75 - smallConstant) * (posRotDifference->y / (upperBounds - lowerBounds))) + 28)) {
+            ret = 1;
+        }
+    }
+    if ((posRotDifference->y <= lowerBounds) && (xzDist <= 28)) {
+        ret = 2;
+    }
+    return ret;
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Stream/func_80B0B81C.s")
+#endif
+
+void EnStream_SuckPlayer(EnStream* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad48;
+    Vec3f posRotDifference;
+    f32 xzDist;
+    f32 yDistWithOffset;
+    s32 pad30;
+    s32 pad2C;
+
+    if (func_80B0B81C(&this->actor.posRot, &player->actor.posRot, &posRotDifference, this->actor.scale.y) != 0) {
+        xzDist = sqrtf(SQ(posRotDifference.x) + SQ(posRotDifference.z));
+        yDistWithOffset = player->actor.posRot.pos.y - (this->actor.posRot.pos.y - 90.0f);
+        player->windDirection = Math_atan2f(-posRotDifference.x, -posRotDifference.z) * 10430.378f;
+        if (3.0f < xzDist) {
+            Math_SmoothScaleMaxMinF(&player->windSpeed, 3.0f, 0.5f, xzDist, 0.0f);
```

Comment:
you may be able to flip both of these comparisons to `var > value` which is a bit easier to read

Reply:
Yep, that works. Thanks for the suggestion, it definitely is easier to read